### DISCUSSION TITLE
[suggestion] variable alias and property alias

### DISCUSSION THREAD

**Discussion by surfsky** [dotnet/csharplang#3508] [Language Ideas]
# 1.Variable alias 

Sometime we have a longvariable name, we can use **variable alias expression** to simply the code. This is example:

``` csharp
var longvariablename = 12;
var txt = match(longvariablename t) // make alias t that can be using in brace
{
    (t<2)  => "Baby";
    (t<10) => "Kids";
    ....
}
```

The variable alias expression may be:
``` csharp
longvariablename t
or
longvariablename as t
or 
longvariablename alias t
```

# 2.Property Alias

Look at the example, MyControl has a property Text referring to child control _block.Text.

``` csharp
public class MyControl
{
    protected TextBlock _block;
    public string Text 
    {
        get => _block.Text;
        set => _block.Text = value;
    }
}
```

Now we can use **property alias expression** to simply code:

``` csharp
public class MyControl
{
    protected TextBlock _block;
    public string Text <=> _block.Text;
    or
    public alias string Text => _block.Text;
}
```

This expression is imported from QML.

...
Maybe we can use this expression to fetch private field. e.g.

``` csharp
private string _text;
public string Text <=> _text;
```

**Comment by MikiraSora** [comment]
You should rename your variable name and make it shorter , not cover bad naming up.

**Comment by surfsky** [comment]
@MikiraSora 
Some times variable name is provide by class property and cannot be changed。eg

``` csharp
while(someobject.SomeProperty t) // make alias t that can be using in brace
{
    // use t 
    ...
}
```

**Comment by hez2010** [comment]
For your case:
```csharp
while(t = someobject.SomeProperty)
{
    // use t 
    ...
}
```

**Comment by svick** [comment]
1. C# already has a syntax to create this kind of an alias: `var t = longvariablename;`. Why does it need another syntax to do the same thing?
2. This is a duplicate of https://github.com/dotnet/csharplang/issues/1007.

**Comment by huoyaoyuan** [comment]
```csharp
using System;
public class C {
    public void M(Class c)
    {
        ref readonly Big b = ref c.LongNameBigField;
    }
}

public struct Big
{
    
}

public class Class
{
    public readonly Big LongNameBigField;
}
```

Event if the variable is a large struct that you don't want to copy it, you can still use ref readonly as an alias.

**Comment by surfsky** [comment]
@hez2010 

```while(t = someobject.SomeProperty)```  will make chaos with == expression.

@svick 

The expression is syntax-sugar to kill some codes. Just like.
``` csharp
string val;
SomeRefFunction(ref val);
->
SomeRefFunction(ref string val);
```

Same the case:

``` csharp
var t = someobject.SomeProperty;
while(t) {.use t..}

->
while(someobject.SomeProperty t) {..use t.}
```
